rfcs/shared/build-system/05.1-react-components-folder-organization.md

```diff
@@ -0,0 +1,41 @@
+# RFC: Group `@fluentui/react-components` Packages to Common Folder
+
+---
+
+Contributors: Fluent UI React Build team, @TristanWatanabe
+
+## Summary
+
+This RFC is a follow-up to the [Repo Folder Organization](https://github.com/JustSlone/fluentui/blob/9e46742b4383af09e58365c0deb7bc70a3297261/rfcs/build-system/05-folder-organization.md) RFC which focuses on the contents of a new common folder for `@fluentui/react-components` packages to live in.
+
+## Background
+
+In prep for the stable release of `@fluentui/react-components`, our repo will start to gain more visitors and contributors who will seek `react-components` related source code and information. Having a common folder where all `@fluentui/react-components` components are located will help make this process straightforward.
+
+## Problem statement
+
+As it currently stands, finding `@fluentui/react-components` related components is not straightforward as they currently live alongside `@fluentui/react`, `@fluentui/northstar` and `@fluentui/web-components` components. Simply searching for the `Button` component will yield a plethora of results spanning all 4 libraries within the monorepo. This can lead to confusion amongst contributors who visit our repo.
+
+## Detailed Design or Proposal
```

Comment:
Have you considered doing something similar to what we've done with v7 where code is maintained on a separate branch? This could maintain the existing file/package structure but all code < v9 would be maintained on a `8.0` branch and removed from `master`.

Reply:
This was actually discussed within the v-build team and ultimately came to the conclusion that branching off v8 would ultimately break github search for that library. So we're going to just restructure the repo folders and unify DX between v8 and v9.

Reply:
This is definitely a valid point but this is more of an incremental step to make it clearer for contributors which packages belong to the v9 library. Branching out should certainly be explored down the line but there's no strong reason to do it right now especially since we're still actively working on both v8 and v9. This is more straightforward for the time being and wouldn't necessarily block future branching attempts (might even make it easier).